core/mybatis-generator-core/src/test/java/org/mybatis/generator/api/dom/java/MethodTest.java

```diff
@@ -0,0 +1,234 @@
+        assertEquals(2, method.getParameters().size());
+        assertEquals("test2", method.getParameters().get(0).getName());
+        assertEquals("test1", method.getParameters().get(1).getName());
+    }
+
+    @Test
+    public void testSetReturnType() {
+
+        Method method = new Method();
+        assertNull(method.getReturnType());
+
+        method.setReturnType(FullyQualifiedJavaType.getIntInstance());
+        assertEquals(FullyQualifiedJavaType.getIntInstance(), method.getReturnType());
+    }
+
+    @Test
+    public void testAddExceptions() {
+
+        Method method = new Method();
+        assertEquals(0, method.getExceptions().size());
+
+        method.addException(new FullyQualifiedJavaType("java.lang.Exception"));
+        assertEquals(1, method.getExceptions().size());
+        assertEquals("Exception", method.getExceptions().get(0).getShortName());
+    }
+
+    @Test
+    public void testSetSynchronized() {
+
+        Method method = new Method();
+        assertEquals(false, method.isSynchronized());
+        method.setSynchronized(true);
+        assertEquals(true, method.isSynchronized());
+    }
+
+    @Test
+    public void testSetNative() {
+
+        Method method = new Method();
+        assertEquals(false, method.isNative());
+        method.setNative(true);
+        assertEquals(true, method.isNative());
+    }
+
+    @Test
+    public void testGetFormattedContent() {
+
+        Method method = new Method("foo");
+        method.setConstructor(false);
+        method.setVisibility(JavaVisibility.PUBLIC);
+        method.setStatic(true);
+        method.setSynchronized(true);
+        method.setFinal(true);
+
+        FullyQualifiedJavaType listType = FullyQualifiedJavaType.getNewListInstance();
+        listType.addTypeArgument(FullyQualifiedJavaType.getStringInstance());
+        FullyQualifiedJavaType typeT = new FullyQualifiedJavaType("T");
+
+        FullyQualifiedJavaType comparatorType = new FullyQualifiedJavaType("java.util.Comparator");
+        comparatorType.addTypeArgument(FullyQualifiedJavaType.getStringInstance());
+        FullyQualifiedJavaType typeR = new FullyQualifiedJavaType("R");
+
+        method.addTypeParameter(new TypeParameter("T", Collections.singletonList(listType)));
+        method.addTypeParameter(new TypeParameter("R", Arrays.asList(listType, comparatorType)));
+
+        FullyQualifiedJavaType functionType = new FullyQualifiedJavaType("java.util.Function");
+        functionType.addTypeArgument(typeT);
+        functionType.addTypeArgument(typeR);
+        method.addParameter(new Parameter(typeT, "t"));
+        method.addParameter(new Parameter(functionType, "func"));
+        method.setReturnType(typeR);
+
+        method.addBodyLine("return func.apply(t);");
+
+        String excepted = "public static final synchronized <T extends List<String>, R extends List<String> & Comparator<String>> R foo(T t, Function<T, R> func) {\r\n"
```

Comment:
You should change \r\n here to System.lineSeparator().  Also on the next line.  I think this is why the build is failing.